eth.c

```diff
@@ -0,0 +1,210 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+// SPDX-License-Identifier: GPL-2.0-only
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include "eth.h"
+#include "wormhole.h"
+
+#define ETH_TIMEOUT_MS 250
+#define ETH_MIN_FW_VERSION 0x6069000
+
+#define ETH_FW_VERSION_ADDR 0x210
+#define ETH_PORT_STATUS_ADDR 0x1200
+#define ETH_NODE_INFO_ADDR 0x1100
+#define ETH_LOCAL_RACK_SHELF_ADDR 0x1108
+#define ETH_LOCAL_NOC0_X_ADDR 0x1110
+#define ETH_LOCAL_NOC0_Y_ADDR 0x1118
+#define ETH_REMOTE_RACK_ADDR 0x1128
+#define ETH_REMOTE_SHELF_ADDR 0x1124
+#define ETH_REQ_WR_PTR_ADDR 0x110a0
+#define ETH_REQ_RD_PTR_ADDR 0x110b0
+#define ETH_REQ_QUEUE_ADDR 0x110c0
+#define ETH_RESP_RD_PTR_ADDR 0x11230
+#define ETH_RESP_WR_PTR_ADDR 0x11220
+#define ETH_RESP_WR_PTR_ADDR 0x11220
+#define ETH_RESP_QUEUE_ADDR 0x11240
+
+#define ETH_CMD_WR_REQ (0x1 << 0)
+#define ETH_CMD_WR_ACK (0x1 << 1)
+#define ETH_CMD_RD_REQ (0x1 << 2)
+#define ETH_CMD_RD_DATA (0x1 << 3)
+
+#define ETH_STATUS_UNKNOWN 0
+#define ETH_STATUS_NOT_CONNECTED 1
+
+static const u32 WH_ETH_NOC0_X[WH_ETH_CORE_COUNT] = { 9, 1, 8, 2, 7, 3, 6, 4, 9, 1, 8, 2, 7, 3, 6, 4 };
+static const u32 WH_ETH_NOC0_Y[WH_ETH_CORE_COUNT] = { 0, 0, 0, 0, 0, 0, 0, 0, 6, 6, 6, 6, 6, 6, 6, 6 };
+
+struct eth_cmd_t {
```

Comment:
Mind doing some one-liners to annotate here as well? I know it's probably just lifted from some existing header files, but we can aim to be a bit nicer to our users. Either oneliners here, or the standard kerneldoc format: https://docs.kernel.org/doc-guide/kernel-doc.html